mod/metrics/set/abstract/metric_answer/validate.rb

```diff
@@ -44,9 +44,10 @@ def validate_category_value value
   end
   self.name = new_name
   detach_subfield(:year)
+  success.year = new_year if success.year
```

Comment:
this makes it look like the year was getting saved correctly but the success wasn't showing it (?).

Doesn't seem like that's the whole problem, right?

Reply:
there are three problems:
1. the form sends the wrong year because javascript fails before it reaches the method that updates the hidden input content tags
2. wagn hitch because the index for card keys saves only 64 characters but the last digit of the year in the example is the 65th character. So the change of the year can't be saved because the key (truncated to 64 characters) is not unique
3. the research page doesn't switch to the changed year